addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -235,7 +235,9 @@ function factory(dependencies) {
          * @returns {mail.message[]}
          */
         async _loadMessages({ extraDomain, limit = 30 } = {}) {
-            this.update({ isLoading: true });
+            if (!this.isLoaded) {
+                this.update({ isLoading: true });
+            }
```

Comment:
The function looks very strange with this: it has become much less declarative with this change.
The underlying issue comes from `loadNewMessages` sharing code with `_loadMessages`, which shouldn't have been the case (clumsy code sharing).

It fixes the issue, and it's minimal, so I'm good with this for the moment. I prefer not cleaning it now, because fixes will take more time and it will slow down fixing this issue and progress of some other tasks.

Reply:
Just move `this.update({ isLoading: true });` into `loadNewMessages` to fix the problem.
Just like `this.update({ isLoadingMore: true });` is in `loadMoreMessages`.

Reply:
But the `if` added in this PR will be necessary no matter what because `postMessage` is calling `loadNewMessages` specifically.

Reply:
I'm thinking `isLoading` must stay where it was actually, it's an important state value related to the loading.
Maybe it's just the condition on the template that is wrong, should not show "loading" if `isLoading` but `isLoaded` has been true before.

Reply:
Actually there are a lot of conditions that depend on `isLoading` for other purpose in thread cache, including determining whether to do the RPC (again) or not, so currently diff that prevents from setting the value to true is definitely not correct.

Reply:
I have moved the condition inside the template.